File: tests/fixtures/cpp/self_class.cpp
// ventouse C++ fixtures — class/struct unit (P1/P2). The constructor is the method named after
// the class; member initialization there is clean.
struct S {
    int n;
    S(int n_) : n(n_) {}          // CPP-CTOR — member init in the ctor -> clean 0
    void set(int v) { n = v; }    // CPP-SET — non-const method writes a member -> dirty 20
    int get() const { return n; } // CPP-GET — const method cannot mutate -> clean 0
};
// S has a dirty method (set) -> the class is dirty: base 10 + set 20 = 30.

File: tests/fixtures/rust/self_class.rs
// ventouse Rust fixtures — struct + impl as the class-like unit (P1/P2).
// Rust has no language constructor: `fn new() -> Self` (building a value) is the "constructor".

struct S {
    n: i32,
}

impl S {
    // RS-NEW — constructor builds/returns a value -> clean 0
    fn new(n: i32) -> Self { S { n } }
    // RS-SET — a &mut self setter writes a field -> dirty 20
    fn set(&mut self, n: i32) { self.n = n; }
    // RS-GET — &self cannot mutate -> clean 0
    fn get(&self) -> i32 { self.n }
}
// S has a dirty method (set) -> the class-like unit is dirty: base 10 + set 20 = 30.
